vault/acl.go

```diff
@@ -47,41 +55,141 @@ func NewACL(policies []*Policy) (*ACL, error) {
 			// Check for an existing policy
 			raw, ok := tree.Get(pc.Prefix)
 			if !ok {
-				tree.Insert(pc.Prefix, pc.CapabilitiesBitmap)
+				var entry aclEntry
+				if pc.MFAMethods != nil && len(pc.MFAMethods) != 0 {
```

Comment:
You can take the `len(nil)` for future reference, so you can simplify to just `len(pc.MFAMethods) != 0`